### PR TITLE
Avoid hard coded icu4c path

### PR DESCRIPTION
**Pull request by tanjialiang**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -386,11 +386,20 @@ endif()
...
-  if(ON_APPLE_M1)
-    list(APPEND CMAKE_PREFIX_PATH "/opt/homebrew/opt/icu4c")
-  else()
+  if (ON_APPLE_M1)
+    execute_process(
+      COMMAND brew --prefix icu4c
+      RESULT_VARIABLE BREW_ICU4C
+      OUTPUT_VARIABLE BREW_ICU4C_PREFIX
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if (BREW_ICU4C EQUAL 0 AND EXISTS "${BREW_ICU4C_PREFIX}")
```
Comment (kgpai) [inline CMakeLists.txt:395]: If this fails we should ensure we append path with "/usr/local/opt/icu4c" .

**Review by kgpai — APPROVED** [review]
cc: @assignUser

File: CMakeLists.txt [hunk 2]
```diff
@@ -387,7 +387,18 @@ set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
...
-    list(APPEND CMAKE_PREFIX_PATH "/opt/homebrew/opt/icu4c")
+    execute_process(
+      COMMAND brew --prefix icu4c
+      RESULT_VARIABLE BREW_ICU4C
+      OUTPUT_VARIABLE BREW_ICU4C_PREFIX
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+    if(BREW_ICU4C EQUAL 0 AND EXISTS "${BREW_ICU4C_PREFIX}")
+      message(
+        STATUS "Found icu4c installed by Homebrew at ${BREW_ICU4C_PREFIX}")
+      list(APPEND CMAKE_PREFIX_PATH "${BREW_ICU4C_PREFIX}")
```
Comment (assignUser) [inline CMakeLists.txt:398]: I think we should move this before `ON_APPLE_M1`  so it's used for both intel and m1 mac and then only distinguish arch's for fallback paths.

**Review by assignUser — COMMENTED** [review]
Thanks good change, a minor comment. Please add a description too.

File: CMakeLists.txt (hunk 2 above)
Comment (kgpai) [reply CMakeLists.txt:398]: cc: @tanjialiang

Comment (tanjialiang) [reply CMakeLists.txt:398]: Saw this after landed. I can have another followup PR to address this. Thanks!

**Pull request closed without merging**